tests/test_api.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+"""
+Functional Test of the API
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import boto3
+import os
+import requests
+import sys
+import unittest
+import uuid
+import json
+from tests import utils
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+import dss
+from tests.infra import DSSAsserts, UrlBuilder
+
+
+class TestApi(unittest.TestCase, DSSAsserts):
+
+    def setUp(self):
+        DSSAsserts.setup(self)
+        self.app = dss.create_app().app.test_client()
+
+    BUNDLE_FIXTURE = 'fixtures/test_api/bundle'  # 's3://hca-dss-test-src/fixtures/test_api/bundle'
+
+    def test_creation_and_retrieval_of_files_and_bundle(self):
+        """Test file and bundle lifecycle.
+        Exercises:
+          - PUT /files/<uuid>
+          - PUT /bundles/<uuid>
+          - GET /bundles/<uuid>
+          - GET /files/<uuid>
+        and checks that data corresponds where appropriate.
+        """
+        bundle = S3TestBundle(self.BUNDLE_FIXTURE)
```

Comment:
Because this is just a fixture and the goal of the test is to test bundle lifecycle, not to exhaustively test the URLs that the system can access.